components/homme/src/theta-l/prim_state_mod.F90

```diff
@@ -976,4 +982,64 @@ subroutine prim_diag_scalars(elem,hvcoord,tl,n,t_before_advance,nets,nete)
  
 
 end subroutine prim_diag_scalars
+
+!doing extrema with level for all elems
+subroutine findExtremaWithLevel(elem,res,field,operation,n0,nets,nete)
+    use kinds, only : real_kind
+    use dimensions_mod, only : np, np, nlev, nlevp
+    implicit none
+    real (kind=real_kind), intent(inout) :: res(1:2) ! extrema and level where it happened
+    character(len=*),      intent(in)    :: field, operation
+    integer,               intent(in)    :: nets,nete,n0
+    type (element_t),      intent(in), target :: elem(:)
+
+    integer                              :: i,j,k,ie,ksize
+    real (kind=real_kind)                :: column(1:nlevp), val, BIGVAL    
+
+    !did not find an approp module to put this in
+    BIGVAL = 1e15
```

Comment:
I recommend against defining BIGVAL and doing the loops below. Just do MINVAL over whatever array slices you need. This routine could probably be ~8 lines if you do that.

Reply:
Since there is also ie index, I don't see how to avoid BIGVAL easily. I shortened the routine as much as I could.